src/components/data/test/DataTable-test.js

```diff
@@ -406,5 +406,81 @@ describe('DataTable', () => {
     });
   });
 
+  context('when table can be ordered', () => {
+    [
+      { direction: 'ascending', expectedArrow: 'ArrowUpIcon' },
+      { direction: 'descending', expectedArrow: 'ArrowDownIcon' },
+    ].forEach(({ direction, expectedArrow }) => {
+      it('shows initial active order', () => {
+        const wrapper = createComponent(DataTable, {
+          order: { field: 'color', direction },
+        });
+        const colorTableCell = wrapper.find('TableCell').at(1);
+
+        assert.equal(colorTableCell.prop('aria-sort'), direction);
+        assert.isTrue(colorTableCell.exists(expectedArrow));
+      });
+    });
+
+    [
+      // Clicking the same column when initially ascending, transitions to descending
+      {
+        initialOrder: { field: 'name', direction: 'ascending' },
+        clickedColumn: 'Name',
+        expectedNewOrder: {
+          field: 'name',
+          direction: 'descending',
+        },
+      },
+      // Clicking the same column when initially descending, transitions to no order
```

Comment:
Was the choice to cycle between `ascending -> descending -> no order` rather than `ascending -> descending -> ascending ...` based on an existing UI? I wonder if it might be better to simplify the behavior so that there is no option to remove the sorting once applied, ie. it would just cycle through ascending/descending.